### PR TITLE
Visitor can view article feature

### PR DESCRIPTION
**Pull request by NiciP**
_As a Visitor
In order to choose an article to read
I would like to be able to see the articles listed on a page_

#### Description
- Add index page to the application
- List articles with title & lead
- Add `Article` model with unit tests
- Add and config article controller
- Add feature test to see title and lead

#### What I have learned working on this feature: 
That we do not need to follow the logic of a journalist creating an article first in order for it to be viewed, it can be created in the database.

### CODE REVIEW

**Review by aleciabenjamin — APPROVED** [review]
lgtm

File: features/step_definitions/basic_steps.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+When("I visit the site") do
+    visit root_path
+end
+
+Given("the following articles exist") do |table|
+    table.hashes.each do |article|
+        Article.create(article)
```
Comment (magnus-thor) [inline features/step_definitions/basic_steps.rb]: Use the FactoryBot for creating objects.

**Review by magnus-thor — CHANGES_REQUESTED** [review]

File: features/step_definitions/basic_steps.rb (hunk 1 above)
Comment (NiciP) [reply features/step_definitions/basic_steps.rb]: ok done that

**Review by magnus-thor — APPROVED** [review]
Looks good

**Pull request merged**